src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/LanguageClient/RazorLSPSMappingServiceTest.cs

```diff
@@ -0,0 +1,156 @@
+        });
+
+        var lazyManager = new Lazy<LSPDocumentManager>(() =>
+        {
+            var manager = new TestDocumentManager();
+            manager.AddDocument(_mockDocumentUri, documentSnapshot.Object);
+
+            return manager;
+        });
+
+        var documentMappingProvider = new LSPDocumentMappingProvider(requestInvoker, lazyManager);
+
+        var service = new RazorLSPMappingService(documentMappingProvider, documentSnapshot.Object, textSnapshot);
+
+        var expectedSpan = _sourceTextRazor.GetTextSpan(mappedRange);
+        var expectedLinePosition = _sourceTextRazor.GetLinePositionSpan(expectedSpan);
+        var expectedFilePath = _mockDocumentUri.LocalPath;
+        var expectedResult = (expectedFilePath, expectedLinePosition, expectedSpan);
+
+        // Act
+        var result = await service.GetTestAccessor().MapSpansAsync(spans, _sourceTextGenerated, _sourceTextRazor);
+
+        // Assert
+        Assert.Single(result, expectedResult);
+    }
+
+    [Fact]
+    public async Task MapSpans_OutsideRange_ReturnsEmpty()
+    {
+        // Arrange
+        var textSpan = new TextSpan(10, 10);
+        var spans = new TextSpan[] { textSpan };
+
+        var documentSnapshot = new Mock<LSPDocumentSnapshot>(MockBehavior.Strict);
+        documentSnapshot.SetupGet(doc => doc.Uri).Returns(_mockDocumentUri);
+        documentSnapshot.SetupGet(doc => doc.Snapshot).Returns(new StringTextSnapshot(_sourceTextRazor.ToString()));
+
+        var textSnapshot = new StringTextSnapshot(s_mockGeneratedContent, 1);
+
+        var textSpanAsRange = _sourceTextGenerated.GetRange(textSpan);
+
+        var requestInvoker = new TestLSPRequestInvoker(new List<(string, object)>()
+        {
+            (LanguageServerConstants.RazorMapToDocumentRangesEndpoint, null)
+        });
+
+        var lazyManager = new Lazy<LSPDocumentManager>(() =>
+        {
+            var manager = new TestDocumentManager();
+            manager.AddDocument(_mockDocumentUri, documentSnapshot.Object);
+
+            return manager;
+        });
+
+        var documentMappingProvider = new LSPDocumentMappingProvider(requestInvoker, lazyManager);
+
+        var service = new RazorLSPMappingService(documentMappingProvider, documentSnapshot.Object, textSnapshot);
+
+        // Act
+        var result = await service.GetTestAccessor().MapSpansAsync(spans, _sourceTextGenerated, _sourceTextRazor);
+
+        // Assert
+        Assert.Empty(result);
+    }
+
+    [Fact]
+    public void MapSpans_GetMappedSpanResults_MappingErrorReturnsDefaultMappedSpan()
+    {
+        // Arrange
+        var sourceTextRazor = SourceText.From("");
+        var response = new RazorMapToDocumentRangesResponse { Ranges = new Range[] { VsLspFactory.UndefinedRange } };
+
+        // Act
+        var results = RazorLSPSpanMappingService.GetMappedSpanResults(_mockDocumentUri.LocalPath, sourceTextRazor, response);
+
+        // Assert
+        Assert.True(results.Single().IsDefault);
```

Comment:
Good point... let me look. This is added because I wanted at least the assertions we had on the old mapping service to apply to the new one. Direct copy (see the `RazorLSPS...` because I only deleted `pan`)